base/array.jl

```diff
@@ -667,6 +677,11 @@ function collect_to!(dest::AbstractArray{T}, itr, offs, st) where T
             return collect_to!(new, itr, i+1, st)
         end
     end
+    lastidx = lastindex(dest)
```

Comment:
This should actually be `last(LinearIndices(dest))`. Will fix in the next round of reviews.